include/spdlog/details/os-inl.h

```diff
@@ -236,8 +236,8 @@ SPDLOG_INLINE size_t filesize(FILE *f)
 #    else
     int fd = ::fileno(f);
 #    endif
-// 64 bits(but not in osx or cygwin, where fstat64 is deprecated)
-#    if (defined(__linux__) || defined(__sun) || defined(_AIX)) && (defined(__LP64__) || defined(_LP64))
+// 64 bits(but not in osx, linux/musl or cygwin, where fstat64 is deprecated)
+#    if ((defined(__linux__) && defined(__GLIBC__)) || defined(__sun) || defined(_AIX)) && (defined(__LP64__) || defined(_LP64))
```

Comment:
Better check for some kind of musl specifc macro to detect it. Who knows what it may break checking __GLIBC__ for this.

Reply:
Glibc defined the macro long time ago and it won’t go away ever however musl does not define any libc identifying macros intentionally

Reply:
I understand , but this would affect any other libc vendors that dont define __GLIBC__ intentionally.

Reply:
its already checking for linux, which limits the C  library choices. Glibc and uclibc already define __GLIBC__, third option is musl and another option is bionic/android which would be similar to musl, if needed it can be controlled  via android define. Do you have any other combination that spdlog supports on linux ?

Reply:
There is no specific combination,  but this is bit dangerous. Who knows what libc impl this would break. According to [this](https://wiki.osdev.org/C_Library#Implementations), there are at least 11. Makes much more sense to check for musl if this is a musl fix

Reply:
Let me put it differently. What this code does currently it glibc specific, LFS64 functions are not posix defined. It’s not Linux specific. So perhaps making the check so that is the right thing here

Reply:
That’s probably right. I am even thinking to remove the use of fstat64 entirely, since it seems that in 64 bit platforms the regular fstat() already returns the 64 version of `off_t` (including android bionic). Could you check ?

Reply:
yes. On 64bit linux and AFAICT BSDs and OSX `off_t` is 64bit